modules/core/04-channel/types/upgrade.go

```diff
@@ -93,13 +94,31 @@ func (u UpgradeError) Error() string {
 	return u.err.Error()
 }
 
+
+// Is returns true if the underlying error is of the given err type.
+func (u UpgradeError) Is(err error) bool {
+	return errors.Is(u.err, err)
```

Comment:
Combining this with `Unwrap` allows us to disregard the `UpgradeError` type wrapper?

Reply:
Would it make sense to actually enforce that the target error is also an `UpgradeError`?

Reply:
I think I would prefer if the `UpgradeError` was effectively identical to other sdk errors as far as the `errorsmod` package is concerned.

I think this might make usage clunky, possibly needing to construct an `UpgradeError` from a specific sdk error in order to do an error comparison. 

```golang
if errorsmod.Is(err, types.UpgradeError{types.ErrInvalidChannel}) {}
```